docs/site/Inside-Loopback-Application.md

```diff
@@ -0,0 +1,173 @@
+  - After the user logs in, the mobile/web app creates a shopping cart for the user.
+
+We have the following API requirements for this usecase:
+
+| Use case              | operations         |  summary             | 
+| --------------------  | --------------------- |----------------------|
+| create a new user     | POST /users           | Basic authentication |
+| query for a user      | GET  /users/{userId}  | query by id          |
+| login as a user       | POST /users/login     | returns a jwt token  |
+
+Similar to this we can breakdown further use cases into API requirements.
+
+- `Shopping use cases`
+  - User starts adding and removing items to the shopping cart.
+  - If the user logs out, the shopping cart is saved and used for the next user session.
+
+- `Order use cases`
+  - User creates an order with items in the shopping cart.
+  - The mobile/web app submits the order and resets the shopping cart.
+  - User takes a look at all the previous orders.
+
+![example shopping](https://github.com/strongloop/loopback4-example-shopping/blob/master/example-shopping.png)
+
+In the example,
+
+- [A product recommendations application](https://github.com/strongloop/loopback4-example-shopping/tree/master/packages/recommender) is defined to mock shopping recommendation APIs. This also gives an apt use case to demonstrate invoking external APIs.
+
+- `MongoDB` is used for saving `User` and `Order` data. This demonstrates CRUD operations on a model.
+
+- `Redis` is used for caching items in the shopping cart. This allows demonstrating use of a KV Connector as well as having multiple datasources.
+
+## API Implementation
+
+The API business logic is separated between various layers in LoopBack:
+
+- [Controllers](https://loopback.io/doc/en/lb4/Controllers.html) represent the `API Endpoints`. These endpoints will have to authenticate incoming request, parse and validate as well as orchestrate calls to `Services` and `Repositories`.
+
+- [Services](https://loopback.io/doc/en/lb4/Calling-other-APIs-and-web-services.html) provide common interfaces for external apis and services. This allows invoking external services without mentioning connection details every time. `Services` interchangeably also provide common interfaces for locally available classes.
+
+- [Models](https://loopback.io/doc/en/lb4/Model.html) and [Relations](https://loopback.io/doc/en/lb4/Relations.html) represent domain objects and provide entity relationship models.
+
+- [Repositories](https://loopback.io/doc/en/lb4/Repositories.html) represent the `Entity layer` for a specific model and handle all CRUD operations on the model. They also use repository of other models to handle `entity relations`.
+
+![Business Logic](imgs/shopping-business-logic.png)
+
+## Exposing the API endpoints
+
+> LoopBack develepers can configure simple controller classes and methods as remote endpoints. Out-of-the-box parsing and validation is done using the provided specification.
```

Comment:
- typo: `develepers`
- Personally, I think the `>` should be a quote which isn't the case here.  But I don't have a better suggestion to indicate this is the main sentence and the paragraph below is the description.

Reply:
agree on the typo ; but the Block Quote = `>` in MD files is a standard highlight option. If there is a better way please let me know. "Quotes" do not get highlighted in MD files.